queens-attack-2.ts

```diff
@@ -0,0 +1,56 @@
+// https://www.hackerrank.com/challenges/queens-attack-2
+
+export function queensAttack(
+  n: number,
+  k: number,
+  rQ: number,
+  cQ: number,
+  obstacles: number[][]
+): number {
+  const moves = {
+    up: n - rQ,
+    down: rQ - 1,
+    left: cQ - 1,
+    right: n - cQ,
+    upRight: Math.min(n - rQ, n - cQ),
+    downRight: Math.min(rQ - 1, n - cQ),
+    downLeft: Math.min(rQ - 1, cQ - 1),
+    upLeft: Math.min(n - rQ, cQ - 1),
+  }
+
+  for (const obs of obstacles) {
+    // obs same row
+    if (obs[0] === rQ) {
+      if (obs[1] > cQ) {
+        moves['right'] = Math.min(moves['right'], obs[1] - cQ - 1)
+      } else {
+        moves['left'] = Math.min(moves['left'], cQ - obs[1] - 1)
+      }
+    }
+    // same column
+    if (obs[1] === cQ) {
```

Comment:
does the logic require `else if` here?

Reply:
you are right, because both cannot be true

Reply:
In these cases, always please use `else if`:
* it improves readability
* it improves performance